apps/api/src/report/report.service.ts

```diff
@@ -148,20 +150,86 @@ export class ReportService {
 
   /**
    * Report 4
+   * @param period report period to select
    * @param f Duration start date YYYY-MM-DD
    * @param t Duration end date YYYY-MM-DD
    * @returns
    */
-  async getLicensingStageApplicants(statuses: Record<string, string>, f: string, t: string) {
-    const { from, to } = this.captureFromTo(f, t);
-    this.logger.log(`getLicensingStageApplicants: Apply date filter from (${from}) and to (${to})`);
+  async getLicensingStageApplicants(period: number) {
+    const report_number = 4;
+    const report = await getRepository(ReportCacheEntity)
+      .createQueryBuilder('rce')
+      .where('rce.report_period = :period', { period })
+      .andWhere('rce.report_number = :report_number', { report_number })
+      .orderBy('rce.created_date', 'DESC')
+      .getOne();
+
+    if (!report) {
+      return [];
```

Comment:
In the normal caching process, shouldn't we try to fetch data if the cache is not found?

Reply:
yep, I will fix this when I add in the option to invalidate the cache